proteus/LinearAlgebraTools.py

```diff
@@ -131,6 +131,70 @@ def save(self, filename):
         _petsc_view(self, filename)
 
 
+class ParVec_petsc4pyTH(p4pyPETSc.Vec):
```

Comment:
I did work on fixing the ParVec support for interleaved and non-interleaved (uniform and mixed) spaces. Blame show's you having added this code though. If it's not being used anywhere then I'm OK just cutting it. We shouldn't need a separate parallel vector for Taylor Hood, just different logic in the main ParVec_petsc4py.